src/GraphQL/Types/Schema.cs

```diff
@@ -34,13 +35,58 @@ public Schema()
         /// <summary>
         /// Create an instance of <see cref="Schema"/> with a specified <see cref="IServiceProvider"/>, used
         /// to create required objects.
+        /// Pulls registered <see cref="IConfigureSchema"/> instances from <paramref name="services"/> and
+        /// executes them.
         /// </summary>
         public Schema(IServiceProvider services)
+            : this(services, true)
+        {
+        }
+
+        /// <summary>
+        /// Create an instance of <see cref="Schema"/> with a specified <see cref="IServiceProvider"/>, used
+        /// to create required objects.
+        /// If <paramref name="runConfigurations"/> is <see langword="true"/>, pulls registered
+        /// <see cref="IConfigureSchema"/> instances from <paramref name="services"/> and executes them.
+        /// </summary>
+        public Schema(IServiceProvider services, bool runConfigurations = true)
+        {
+            _services = services;
+
+            Directives = new SchemaDirectives();
+            Directives.Register(Directives.Include, Directives.Skip, Directives.Deprecated);
+
+            if (runConfigurations)
+            {
+                if (services.GetService(typeof(IEnumerable<IConfigureSchema>)) is IEnumerable<IConfigureSchema> configurations)
```

Comment:
Why `services.GetService(typeof(IEnumerable<IConfigureSchema>)) is IEnumerable<IConfigureSchema> configurations)` ?

`services.GetServices<IConfigureSchema>()`. When one registers multiple `IA` in DI container then you pull `IA` from it, not `IEnumerable<IA>`.

Reply:
No, to retrieve multiple of the same type you have to request an IEnumerable - at least with MS DI. The MS DI will give you a list of registrations. If you only request the specific type you will only get the last one registered.

Null is allowed to be returned if none are registered (not sure but MS DI might actually return an empty list). So the `is` check both performs the cast and also checks if not null at the same time.

Reply:
I didn’t see at first that you wrote GetServices. 

`GetServices<T>` is probably an extension method that is not available without the MS DI Abstractions package. But all it will do is request an IEnumerable of the specified type. There is only one method on IServiceProvider - `GetService(Type)`

Reply:
(And GraphQL.NET still does not use the MS DI abstractions nuget package)

Reply:
Right, right. I did not think at all. Yes, these methods are extensions and MSDI specific.
```
  public static IEnumerable<T> GetServices<T>(this IServiceProvider provider)
        {
            if (provider == null)
            {
                throw new ArgumentNullException("provider");
            }

            return provider.GetRequiredService<IEnumerable<T>>();
        }
```